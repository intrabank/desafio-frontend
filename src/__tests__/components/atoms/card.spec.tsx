import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import Card from '../../../components/atoms/card';
import renderer from 'react-test-renderer';

describe('Card', () => {
	it('should render card component', () => {
		const tree = renderer.create(<Card>Test</Card>).toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('should render multiple children', () => {
		render(
			<Card>
				<h1>Test</h1>
				<p>Test 2</p>
				<button>Test 3</button>
			</Card>
		);

		expect(screen.getByText('Test')).toBeInTheDocument();
		expect(screen.getByText('Test 2')).toBeInTheDocument();
		expect(screen.getByText('Test 3')).toBeInTheDocument();
	});
});
