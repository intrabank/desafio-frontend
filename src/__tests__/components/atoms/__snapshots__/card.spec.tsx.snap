// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should render card component 1`] = `
<div
  className="c-iRHVgi"
>
  Test
</div>
`;
